sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -427,8 +427,12 @@ case class MapObjects private(
       case _ => ""
     }
 
+    val inputDataType = inputData.dataType match {
+      case p: PythonUserDefinedType => p.sqlType
```

Comment:
I think it's fixed by https://github.com/apache/spark/pull/13778/files#diff-47e9c0787b1c455e5bd4ad7b65df3436R209 . Can you double check it? If we revert this change, will test fail again?